rct229/ruleset_functions/get_hw_loop_zone_list_w_area.py

```diff
@@ -0,0 +1,91 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_heating_type_fluid_loop import (
+    is_hvac_sys_heating_type_fluid_loop,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_hvac_sub_functions.is_hvac_sys_preheating_type_fluid_loop import (
+    is_hvac_sys_preheating_type_fluid_loop,
+)
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import (
+    find_exactly_one_hvac_system,
+)
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ZERO, pint_sum
+
+HEATING_SOURCE = schema_enums["HeatingSourceOptions"]
+
+
+def get_hw_loop_zone_list_w_area(rmi_b):
+    """
+    Get the list of zones and their total floor area served by each HHW loop in a baseline ruleset model instance.
+
+    Parameters
+    ----------
+    rmi_b: The baseline ruleset model instance that needs to get the list of zones with their total floor area served by each HHW loop.
+
+    Returns hw_loop_zone_list_w_area_dictionary A dictionary that saves the list of zones and the total floor area
+    served by each HHW loop, i.e. {loop_1.id: {"ZONE_LIST": [zone_1.id, zone_2.id, zone_3.id], "TOTAL_AREA": 10000},
+    loop_2.id: {"ZONE_LIST": [zone_10.id], "TOTAL_AREA": 500}}
+
+    -------
+    """
+    hw_loop_zone_list_w_area_dict = dict()
+    for zone in find_all("$..building_segments[*].zones[*]", rmi_b):
+        # None or empty will be falsy
+        zone_area = pint_sum(
+            [
+                space.get("floor_area", ZERO.AREA)
+                for space in find_all("$.spaces[*]", zone)
+            ]
+        )
+        for terminal in find_all("$.terminals[*]", zone):
+            hhw_loop_id = None
+            if terminal.get("heating_source") == HEATING_SOURCE.HOT_WATER:
+                # if terminal has reheat.
+                # If a terminal heating source is hot water, then it must have a heating_from_loop
+                # Otherwise, raise exception
+                hhw_loop_id = getattr_(terminal, "terminal", "heating_from_loop")
+            if not hhw_loop_id and terminal.get(
```

Comment:
If the condition on line 43 is True, then hhw_loop_id will be set, so line 48 could be
```
elif teminal.get("served_by_heating_ventilation_air_conditioning_system"):
```